packages/data-stores/src/user/actions.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Internal dependencies
+ */
+import { ActionType, CurrentUser, NewUserResponse, CreateAccountParams } from './types';
+
+export const receiveCurrentUser = ( currentUser: CurrentUser ) => ( {
+	type: ActionType.RECEIVE_CURRENT_USER as const,
```

Comment:
I'm not a fan of having to specify `as const` in all these action creators but I think this pattern was copied from other data stores. I might have to come up with a proposal for how we can rework the types if I want it to be something different :)

Reply:
Without this annotation, the type system just knows we have a `string`:
![no const](https://user-images.githubusercontent.com/841763/72604381-13f5d680-391b-11ea-81b8-b85e339af26a.png)

With the annotation, we get the string constant `RECEIVE_CURRENT_USER`. I've been tempted to add another `const` on the object so it becomes readonly 😅:
![const](https://user-images.githubusercontent.com/841763/72604380-135d4000-391b-11ea-89ba-46a24bb3bacf.png)

Again, I'm open to trying different things.

Reply:
Thanks @sirreal. The alternatives—now that I've re-examined them—aren't as good as I remember :) Either we just use string literals for action types everywhere and defined action interfaces up front (not a bad approach, has pros and cons); or we used classes as action creators. I can't find the Medium post where the idea originally came from, [but it looks like this](https://gist.github.com/p-jackson/1ca023614930f4fcf89868ea216c2335).

The action-creators-as-classes method is kinda cool, but is another thing for devs to learn, and it doesn't look like our other JS Redux code. I like the idea of keeping the onramp to TS as smooth as possible. Plus it requires a (simple) middleware since Redux only accepts POJO actions out of the box.

So I've talked myself into thinking that the `as const` is probably the best of all the worlds.

Another way to const-ify :all-the-things: is to use the `<const>` notation:

```typescript
function createActionConst( value: number ) {
	return <const>{
		type: 'MY_TYPE',
		value
	};
}
```

That seems kinda nice.